src/crafting.rs

```diff
@@ -231,13 +237,37 @@ pub fn calculate_crafting_profit(
     let mut total_crafting_cost = Rational32::zero();
     let mut crafting_count = 0;
     let mut total_crafting_steps = Rational32::zero();
+    let mut used_recipes = HashMap::new();
 
     let mut min_sell = 0;
     let max_sell = tp_listings_map.get(&item_id)
         .unwrap_or_else(|| panic!("Missing listings for item id: {}", item_id))
         .buys.last().map_or(0, |l| l.unit_price);
     let mut breakeven = Rational32::zero();
 
+    // Check if we can craft this
+    let mut used_recipe = |recipe: &Recipe| {
+        if let Some(id) = recipe.id {
+            if !&used_recipes.contains_key(&id) {
+                match recipe.source {
+                    // Not included in the API
+                    RecipeSource::Automatic | RecipeSource::Discovered => {
+                        // TODO: check if account has a char with the required crafting level
+                        used_recipes.insert(id, true);
+                    }
+                    _ => {
+                        if let Some(known_recipes) = known_recipes {
+                            used_recipes.insert(id, known_recipes.contains(&id));
+                        } else {
+                            // If we have no known recipes, assume we know none
+                            used_recipes.insert(id, false);
+                        }
+                    }
+                }
```

Comment:
Do you mean that the recipes are not included in the api, because they can't be purchased? If that's the case would it make sense to just check `Purchased`?

```suggestion
                match recipe.source {
                    RecipeSource::Purchased => {
                        if let Some(known_recipes) = known_recipes {
                            used_recipes.insert(id, known_recipes.contains(&id));
                        } else {
                            // If we have no known recipes, assume we know none
                            used_recipes.insert(id, false);
                        }
                    }
                    // Not included in the API
                    _ => {
                        // TODO: check if account has a char with the required crafting level
                        used_recipes.insert(id, true);
                    }
                }
```

Reply:
Some are awarded by an achievement instead. I've make it more explicit by specifying all branches of the enum.